biogeophys/FatesPlantRespPhotosynthMod.F90

```diff
@@ -434,48 +464,70 @@ subroutine FatesPlantRespPhotosynthDrive (nsites, sites,bc_in,bc_out,dtime)
-                                 
+                                 ! Approach taken from ELM carbon-only photosynthesis code (Q. Zhu)
+                                 ! using three Rubisco enzyme terms related to leaf N and lnc at top of the canopy
+                                 vcmax25top_prt = lnc_top * flnr(ft) * fnr * act25
+    
+                                 kn_prt = decay_coeff_kn(ft,vcmax25top_prt)
+                                 nscaler_prt = exp(-kn_prt * cumulative_lai)
+
                               case (prt_cnp_flex_allom_hyp)
 
                                  leaf_c  = currentCohort%prt%GetState(leaf_organ, all_carbon_elements)
                                  if( (leaf_c*slatop(ft)) > nearzero) then
                                     leaf_n  = currentCohort%prt%GetState(leaf_organ, nitrogen_element)
+                                    leaf_p  = currentCohort%prt%GetState(leaf_organ, phosphorus_element)
                                     lnc_top = leaf_n / (slatop(ft) * leaf_c )
+                                    lpc_top = leaf_p / (slatop(ft) * leaf_c )
+                                    ! lnc_top = leaf_n / currentPatch%tlai_profile(cl,ft,iv) !testing with LAI method
+                                    ! lpc_top = leaf_p / currentPatch%tlai_profile(cl,ft,iv) !testing with LAI method
+                                    lnc_top = min(max(lnc_top,0.25_r8),3.0_r8) !based on doi: 10.1002/ece3.1173
+                                    lpc_top = min(max(lpc_top,0.014_r8),0.85_r8) !based on doi: 10.1002/ece3.1173
                                  else
-                                    lnc_top  = prt_params%nitr_stoich_p1(ft,prt_params%organ_param_id(leaf_organ))/slatop(ft)
+                                    lnc_top = prt_params%nitr_stoich_p1(ft,prt_params%organ_param_id(leaf_organ))/slatop(ft)
+                                    lpc_top = prt_params%phos_stoich_p1(ft,prt_params%organ_param_id(leaf_organ))/slatop(ft)
                                  end if
                                     
                                  ! If one wants to break coupling with dynamic N conentrations,
                                  ! use the stoichiometry parameter
                                  ! lnc_top  = prt_params%nitr_stoich_p1(ft,prt_params%organ_param_id(leaf_organ))/slatop(ft)
                                  
+                                 vcmax25top_prt = exp(vcmax_np1(ft) + vcmax_np2(ft)*log(lnc_top) + &
+                                          vcmax_np3(ft)*log(lpc_top) + vcmax_np4(ft)*log(lnc_top)*log(lpc_top))
+                                 jmax25top_prt = exp(jmax_np1 + jmax_np2*log(vcmax25top_prt) + jmax_np3*log(lpc_top))
+                                 vcmax25top_prt = min(max(vcmax25top_prt, 10.0_r8), 150.0_r8)
+                                 jmax25top_prt  = min(max(jmax25top_prt, 10.0_r8), 250.0_r8)
+
+                                 kn_prt = decay_coeff_kn(ft,vcmax25top_prt)
```

Comment:
are the kn_prt and nscaler_prt variables used @jenniferholm ?  Should we remove these?